### DISCUSSION TITLE
Account Leasing - A novel proposal to the problem of account reserves

### DISCUSSION THREAD

**Discussion by xrpl365** [XRPLF/XRPL-Standards#238] [XLS Ideas (pre standard proposal)]
# Account Leasing

```
Title:       Account Leasing
Type:        Draft
Revision:    1
Authors:     Chris Dangerfield @xrpl365
Affiliation: XRPL365 Ltd
Reviewed by: Denis Angell
```

## Abstract

The Account Reserve Leasing proposal is a new innovative approach designed to overcome the challenges posed by account reserve requirements on the XRP Ledger (XRPL). By enabling investors to fund pools of XRPL accounts and allowing businesses to lease these accounts for their users, the proposal lowers barriers to entry, making it easier for new users and projects to join the ecosystem. This model not only ensures a secure return for investors but also incentivizes businesses to drive user engagement, ultimately facilitating greater adoption and supporting the growth of projects within the XRPL ecosystem.

## Background

Account reserves on the XRPL are crucial for spam prevention and network security, despite the challenges they pose for onboarding new users. While some may argue for lowering these reserves to simplify onboarding, doing so would undermine their essential role in maintaining the integrity of the XRPL. Over time, several XLS proposals have attempted to address the onboarding challenges created by account reserves, but none have fully succeeded due to falling short of solving the entire problem in one way or another.

This proposal introduces an innovative approach that not only acknowledges the necessity of account reserves, but embraces them, making them a net positive for the ecosystem. Building on the foundational work of past XLS proposals, this proposal combines some of the key principles from previous efforts with a strategy that embraces the existence of reserves. By doing so, it offers a practical and potentially scalable solution that aligns the interests of investors, businesses, and users, while promoting broader adoption and engagement within the XRPL ecosystem.

To be clear whilst the term proposal is used throughout this document, this is presented as an Idea, not an XLS. I beleive the fundamental prinipcles of this proposal are sound and whilst many of the technicals have been thought through, I stopped short of creating an actual spec given this is very much blue-sky thinking at this point, but I look forward to constructive feedback and the evolution of this idea.

## How do I judge success versus other proposed solutions

To be deemed a success a solution must:

- Not lower the base reserve or materially change how reserves function.
- Remove or significantly reduce the barrier to entry for users and business.
- Be implementable, without adding significant burden to the ledger.

## The purpose of Account Reserves

Account reserves on the XRPL are a necessary safeguard designed to protect the network from spam, abuse, and excessive resource consumption. The core idea behind account reserves is to ensure that the ledger remains scalable, efficient, and secure.

## Problems created by Account Reserves

While the account reserve mechanism effectively secures the network, it creates barriers to entry, particularly for new users or projects with large onboarding requirements, limiting accessibility and discouraging widespread adoption in some cases. This is considered by many to be the single biggest challenge to mass retail adoption.

## Core Idea

The leasing model allows investors to invest into a pool of fully funded XRPL accounts. These accounts are then rented out to businesses that need to provide their customers with XRPL accounts without the burden of pre-funding them. The system is structured to ensure that investors receive a return on their investment, businesses reduce their onboarding costs, and users gain access to fully functional accounts without upfront payments. Additionally, businesses eventually regain the rental costs by assuming the claim after the investor’s has been repaid, further incentivising legitimate usage.

## How It Works

**Creation of the Account Pool**

- Investors contribute funds to create a pool of fully funded XRPL accounts. Investors have a claim on the accounts as collateral, ensuring they receive a proportional share of the returns generated by the pool. While the investor's claim secures their investment, they do not own the account itself—ownership remains with the user, who holds the private key. However, the user is restricted from deleting the account and withdrawing the reserve until full repayment (including the rental cost) is completed. This structure ensures that the account remains active and the investors' interests are protected.

**Account Rental by Businesses**

- Businesses that need to onboard users can rent these fully funded accounts from the pool. This rental agreement allows the business to provide new users with a functional XRPL account without needing to allocate significant capital upfront. Instead, the business agrees to pay a monthly fee of 1.125 XRP for each account they rent, ensuring the account contributes to the pool's income, even if it is not used. The business is incentivized to get its users to participate in the network by generating transactions, as this will contribute towards account rental cost.

**User Experience**

- Users receive the rented account and can begin using it immediately without any upfront cost. While the account is in this rented state, the user cannot delete the account and recover the reserve; instead, the reserve would be redirected to the pool if the account is deleted. The user conducts transactions with each transaction incurring an additional fee, on top of the network fee, of 250,000 drops (approximately $0.15). These higher fees are designed to contribute towards paying off the account’s reserve faster.

**Repayment and Yield Generation**

- The fees collected from user transactions and the monthly fees paid by the business accumulate over time. The goal is to generate enough income to cover the 10 XRP reserve plus a 35% return (totaling 13.5 XRP) within a set timeframe, typically one year.
- As users engage more with their accounts, they help accelerate the repayment process. For businesses, this means that encouraging user activity reduces the financial burden of the monthly fees, as more of the account’s cost is covered by the transaction fees.

**Transition of the Claim to the Business**

- Once the account has generated the required 13.5 XRP, the pool’s claim on the account is removed. At this point, the business that rented the account assumes the claim and the transaction fees flow back to the business to a maximum value of what the business paid in rental fees. The user continues to pay the increased transaction fees until the business contribution has been repaid. This ensures that the business eventually recoups their rental payments and ensures that bad actors cannot gain access to low-cost accounts to later abuse once the business has paid for the account reserve.

**Transfer of Account Ownership**

- After the user has repaid the full 10 XRP reserve (and the business' rental fees, whichever is higher) through increased fees, the business’s claim is removed, and the user can recover the accounts reserve in the normal way and the increased transaction fees no longer apply.

**Default and Account Reclamation**

- If the business defaults on the monthly payments, the pool has the right to reclaim the account’s reserve to protect the investment. Upon default, the 10 XRP (NOT 8) reserve is reclaimed by the pool, effectively removing the reserve from the account. While this action renders the account unusable for transactions (assuming it had no other balance), it does not impact any other assets held within the account. The user can regain full functionality of the account by refunding the reserve themselves. Once the reserve is restored, the user retains complete control over the account, free from any external claims. This mechanism ensures that the investors are safeguarded from potential losses while allowing users the opportunity to regain access to their account by restoring the reserve.
- If the business deems the user is no longer viable for them, they can withdraw their sponsorship by stopping to make the monthly payment and the pool will claim back the account reserve and keep any rental income to-date.
- Once a business owns the claim, it is possible that the user may never use the account and be deemed low-value. As the entity with the claim on the account, they have the right to remove the reserve but only to the value of their contribution less any transaction fees paid to them since they took ownership of the claim, meaning the account will likely exist with a small amount of XRP, less than the reserve but greater than zero; this is to ensure that a bad actor with a business cannot abuse the system and claim more XRP back from the account than they themselves have contributed; any XRP paid in transaction fees remains the property of the account owner.

## Incentives and Benefits

**For Businesses**

- Cost Efficiency: Businesses can onboard users without a large upfront investment, spreading the cost over time through manageable monthly payments. This makes it easier to scale user acquisition without straining cash flow.
- Increased Engagement: The model encourages businesses to drive user engagement, as higher user activity reduces the overall cost by accelerating the repayment process. Additionally, the business benefits from eventually regaining their contribution, recouping their rental payments, through incentivization for continued account usage.

**For Users**

- Choice: Users have a choice to fund their own account as they do now or opt to use a leased account rented to them by the pool and sponsored by the business who is onboarding them.
- No Upfront Cost: Users receive a fully functional XRPL account without having to pay the initial reserve, lowering the barrier to entry.
- Path to Ownership: Users are incentivized to use their accounts actively, knowing that consistent use will eventually lead to full ownership of the account. meaning the removal of the high transaction fee and ability to reclaim the reserve.

**For Investors**

- Guaranteed Return: Investors are assured of a return on their initial investment, with the added potential for a 35% return over the year.
- Risk Mitigation: The ability to reclaim accounts in the event of a business default adds a layer of security to the investment, reducing the risk of loss.

## Basic Flows

**Onboarding**

- Bob owns a business that provides a cool service.
- Alice wants to use Bob's service.
- Bob's app onboard's Alice and gives her a choice on sign-up.
  - Pay a setup fee which covers the cost of account activation.
    - Alice pays the setup fee and the flow is unchanged.
  - Use a leased account.
    - Alice chooses no setup fee and agrees to terms of use with Bob.

**Day-to-Day (simple)**

- Bob completes onboarding and is charged 1.125XRP for the first month's rental.
- Alice makes her first transaction and pays an additional transaction fee of 250,000 drops.
- During the next 28 days, Alice uses Bob's service 3 more times and pays an additional 750,000 drops in fees.
- 28 days later, Bob pays another 1.125 XRP for the 2nd month's rent.
- Alice does another 12 transactions, paying 3 XRP in fees.
- Bob pays another 1.125 XRP for a 3rd time and Alice makes 14 transactions.
- Bob pays another 1.125 XRP for the 4th time and Alice makes 10 transactions.
  - The first 6 of Alice's transactions complete the pay-off of the account and the pool releases the hold on the account.
  - The ledger determines that Bob has supported the account to the tune of 4.5 XRP and the claim flows to Bob.
  - Alice's final 4 transaction fees find their way to Bob; Bob is now owed 3.5 XRP.
- Alice makes 20 more transactions the following month.
  - The first 14 transactions incur the 250,000 drop fee and this goes to Bob.
  - Bob is now fully repaid and the ledger releases Bob's claim on the account.
  - Alice owns the account and has essentially earned the account reserve sitting on the account.
  - The last 6 transactions occur with zero rental fees.

**Summary**

- Chris, who was an investor has earned his share in the 3.5 XRP rental return; never in fear of losing his original investment as the locked account reserve is the loan collateral.
- Bob has a new customer, who is actively using his service having never had to pay a setup fee. Bob never had to find the upfront capital to onboard Alice. He was able to cover the monthly rental fees from the fees his platform earned from Alice's usage.
- Alice had a great user experience, wasn't put off by an initial setup fee and now has her own account to which she always had 100% self-custody without paying a reserve, but now owns the reserve.

## Potential Challenges

**System Exploitation**

- This feature cannot be exploited to gain access to free or low-cost accounts. Businesses cannot claim more XRP than they contribute.

**Regulatory Compliance**

- The yield-generating aspect of the model could attract regulatory scrutiny and will need to be considered carefully. However, at its core this is a simple lease agreement.

#### How does Account Reserve Leasing benefit users?

Users benefit from this model as they receive fully functional XRPL accounts without needing to pay the initial reserve fee. This lowers the barrier to entry and encourages new users to join the ecosystem. Over time, as users conduct transactions, they contribute to paying off the account reserve, ultimately leading to full ownership of the account.

#### What advantages does this model offer businesses?

Businesses can onboard users without significant upfront costs by leasing XRPL accounts. The model also incentivizes businesses to encourage user engagement, as higher user activity helps to repay the reserve faster, reducing the financial burden of the monthly lease payments. Eventually, businesses can recoup their rental payments, making this a cost-efficient solution for user acquisition.

#### How do investors earn returns in this system?

Investors earn returns through the monthly fees paid by businesses for leasing the accounts, as well as the transaction fees incurred by users. The model offers a 35% return on the initial reserve over a typical timeframe of one year, with the added security of reclaiming the reserve in case of a business default; it’s a super safe investment.

#### How can the investor be assured their investment cannot go down?

The investor's investment is secured by the XRPL protocol, which requires each account to maintain a minimum reserve balance. This reserve acts as collateral for the investment and is locked by the protocol, ensuring it cannot be accessed by any external party. If a business defaults, the reserve remains intact and can be reclaimed by the pool, guaranteeing that the investor’s principal is protected and cannot go down.

#### What happens if a business defaults on its lease payments?

If a business defaults on its lease payments, the pool has the right to reclaim the account's reserve, ensuring their investment is protected. The user can regain full control of the account by restoring the reserve themselves, but until then, the account remains inactive.

#### How are transaction fees handled in this model?

While using a leased account, users incur an additional transaction fee of 250,000 drops (approximately $0.15) per transaction, on top of the standard network fee. These additional fees help pay off the account’s reserve faster. Once the reserve and business rental fees are fully repaid, the increased transaction fees are removed, and the claim on the account is released.

#### Can this model be exploited for free or low-cost accounts?

No, the system is designed to prevent exploitation. Businesses cannot claim more XRP than they have contributed, and users must repay the full reserve and any associated fees before gaining access to any locked reserves. This ensures that all participants contribute fairly to the ecosystem.

#### What happens after the rental fee is fully repaid?

After the reserve (including the 35% return) is fully repaid through transaction fees, the pool's claim on the account is released. The business that rented the account assumes the claim and can continue to receive transaction fees until their rental payments are fully recovered. After that, the user gains full ownership of the account, and the increased transaction fees are lifted.

#### Are there any regulatory concerns with this model?

The yield-generating aspect of Account Reserve Leasing could attract regulatory scrutiny, particularly regarding securities laws. Careful consideration and compliance with relevant regulations will be necessary to ensure the model operates within legal boundaries.

#### How does this model impact the overall growth of the XRPL ecosystem?

By lowering the barriers to entry and making it easier for businesses to onboard users, Account Reserve Leasing promotes broader adoption of the XRPL. The model aligns the interests of investors, businesses, and users, encouraging legitimate use and long-term engagement, which supports the sustainable growth of the XRPL ecosystem.

#### What are the risks associated with this model?

The primary risks include potential regulatory challenges and the possibility of business defaults. However, the model is designed with safeguards, such as the ability to reclaim reserves in the event of a default, to mitigate these risks and protect investors' interests without overly harming the account owner.

#### Can users eventually own the accounts they are using?

Yes, users can eventually gain full ownership of the accounts they are using. Once the reserve and business rental fees are fully repaid through transaction fees, the increased fees are lifted, and users can recover the account reserve as normal.

#### How does this model ensure the security of user accounts?

The security of user accounts is maintained by ensuring that users always hold the private key to their accounts. However, they are restricted from deleting the account and reclaiming the reserve until the full repayment is completed. This structure protects investors' interests while ensuring users maintain control over their accounts.

#### What incentives are in place for businesses to encourage user activity?

Businesses are incentivized to encourage user activity because higher user engagement accelerates the repayment of the reserve. This reduces the financial burden of the monthly lease payments and allows businesses to recoup their rental payments more quickly.

#### What happens if a user stops using the account?

If a user stops using the account, the business sponsoring the account can choose to stop making the monthly lease payments, which would lead to the pool reclaiming the account’s reserve. The user would need to refund the reserve to regain full access to the account.

#### How can this model contribute to the wider adoption of the XRP Ledger?

By reducing the upfront costs associated with account creation, this model makes it easier for new users and projects to join the XRPL ecosystem. This increased accessibility can drive broader adoption and support the growth of the XRP Ledger by making it more inclusive and user-friendly.

## Technical considerations - work in progress

- Account Root needs a new field to record the status so that it can be treated differently and it needs to record the claim account.
- Account deletion function must now change the flow of the reserve to the claim account when leased.
- Account deletion policy needs updating to recognize the current 2 XRP burn is not applicable to leased accounts.
- Every TX will need to check if the account is leased and there will need to be the new fee applied - this will presumably result in a second payment transaction - Maybe speak to tequ about how he was gonna handle this? Was thinking maybe like emitting a new TX in a hook, but implicitly signed. Will need to pull the destination from the Account claim field.
- Automated business payments are needed, we cannot rely on the business to send money, it needs to be pulled. This could be built or we could build subscription service to be re-usable by this proposal.
- Failed business payments need handing and ideally this needs some logic, like a 2nd attempt or similar as to initiate the reserve claw-back due to a low balance is probably not good.
- Rental payment tracking is an essential part of the proposal, we have to track this and I expect store in account state, so its fast to access when needed.
- Automation of claims, transfer of claim ownership and eventual claim removal remain a challenge to be over-come.
- We should probably provide business with a way to terminate, not just default as we may need to design default protection.
- Transactions fees are currently fixed at 250,000 drops but there is an argument that maybe this could be dynamic based on an algo, so first tx is 600k, 2nd 400k, 3rd 200k or some similar calc to ensure low usage accounts contribute more initially - needs thought.
- Reserve claim is going to need to be aware of the current state of the least and who has paid what to ensure the rules are applied correctly.
- Need to consider any regulatory requirements for investors to see required info, such as the state of the pool, how many accounts, percentages of yield earned, no idea what is needed by will need querying consideration.
- What is required regarding SDKs and Wallet support.
- There is a current path where this could be (not exploited) as no-oe can gain, but there is a an obscure route that can be taken to essentially drain the reserve,leaving the investor out of pocket, but this risk can be mitigated with a code change. I'd be interested to see how many readers can identify it OR in fact some other issue tha maybe I have not considered. I think the point is that this is an idea and I acknowledge there is work to be done, but in my view if the principle of the idea is supported, it warrants the work, because the prize is huge.

## Conclusion

This proposal presents an innovative and pragmatic approach to addressing the challenges posed by the XRPL's account reserve requirements. By creating a system where investors, businesses, and users are all incentivized to participate and contribute, this model not only lowers the barriers to entry but also ensures that the network remains secure, scalable, and efficient. Investors are guaranteed returns with mitigated risk, businesses can onboard users without significant upfront costs, and users gain access to fully functional accounts without the need for immediate funding.

This system is carefully designed to prevent exploitation, ensuring that no party can gain undue advantage. The continued presence of increased transaction fees until the reserve and business rental fees are fully repaid ensures that every participant in the network has a stake in its success, promoting legitimate use and long-term engagement.

Moreover, while the yield-generating aspect of the model may require careful regulatory consideration, the benefits of reduced onboarding friction, enhanced network participation, and the potential for broader adoption of the XRPL far outweigh the challenges. By aligning the interests of all parties involved, this model fosters a more inclusive and resilient ecosystem, paving the way for the sustainable growth of the XRPL.

**Comment by mvadari** [comment]
You still have to source XRP for transaction fees, so how much does this really help with the friction of account creation?

[a comment by an automated account is not shown]
